### ISSUE TITLE
Torrents with apostrophe's in the title having issues?

### ISSUE THREAD

**Issue by SaskiFX** [Jackett/Jackett#328]
I have a setup with HD Torrents, and I am encountering an issue with Sonarr. When I tell Sonarr to download this episode, I get an error getting the file from jackett. 

```
May 01, 2016 07:51:09 PM    Info    

ReflectedHttpActionDescriptor ExecuteAsync Invoking action 'GetLogs()'

May 01, 2016 07:51:09 PM    Info    

ApiControllerActionInvoker InvokeActionAsync Action='GetLogs()'

May 01, 2016 07:51:09 PM    Info    

JackettAPINoCacheAttribute OnActionExecutingAsync 

May 01, 2016 07:51:09 PM    Info    

JackettAPINoCacheAttribute OnActionExecutingAsync Action filter for 'GetLogs()'

May 01, 2016 07:51:09 PM    Info    

HttpActionBinding ExecuteBindingAsync 

May 01, 2016 07:51:09 PM    Info    

HttpActionBinding ExecuteBindingAsync 

May 01, 2016 07:51:09 PM    Info    

JackettAuthorizedAttribute OnAuthorizationAsync 

May 01, 2016 07:51:09 PM    Info    

JackettAuthorizedAttribute OnAuthorizationAsync 

May 01, 2016 07:51:09 PM    Info    

ApiControllerActionSelector SelectAction Selected action 'GetLogs()'

May 01, 2016 07:51:09 PM    Info    

ApiControllerActionSelector SelectAction 

May 01, 2016 07:51:09 PM    Info    

AdminController ExecuteAsync 

May 01, 2016 07:51:09 PM    Info    

HttpControllerDescriptor CreateController Jackett.Controllers.AdminController

May 01, 2016 07:51:09 PM    Info    

DefaultHttpControllerActivator Create Jackett.Controllers.AdminController

May 01, 2016 07:51:09 PM    Info    

DefaultHttpControllerActivator Create 

May 01, 2016 07:51:09 PM    Info    

HttpControllerDescriptor CreateController 

May 01, 2016 07:51:09 PM    Info    

DefaultHttpControllerSelector SelectController Admin

May 01, 2016 07:51:09 PM    Info    

DefaultHttpControllerSelector SelectController Route='MS_SubRoutes:System.Web.Http.Routing.IHttpRouteData[]'

May 01, 2016 07:51:09 PM    Info    

  http://knox-srv.duckdns.org/Admin/GetLogs?_=1462146658180

May 01, 2016 07:51:06 PM    Info    

DownloadController Dispose 

May 01, 2016 07:51:06 PM    Info    

DownloadController Dispose 

May 01, 2016 07:51:06 PM    Info    

  Content-type='none', content-length=unknown

May 01, 2016 07:51:06 PM    Info    

DownloadController ExecuteAsync 

May 01, 2016 07:51:06 PM    Info    

JackettAPINoCacheAttribute OnActionExecutedAsync 

May 01, 2016 07:51:06 PM    Info    

JackettAPINoCacheAttribute OnActionExecutedAsync Action filter for 'Download(String indexerID, String path, String apikey, String file)'

May 01, 2016 07:51:06 PM    Info    

ApiControllerActionInvoker InvokeActionAsync 

May 01, 2016 07:51:06 PM    Info    

ReflectedHttpActionDescriptor ExecuteAsync Action returned 'StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: <null>, Headers:
{
}'

May 01, 2016 07:51:06 PM    Error   

Error downloading hdtorrents http://hdts.ru/download.php?id=69aca296b4b894c1d99bad9f3c3f237bace36d5c&f=Criminal+Minds+S11E21+Devil's+Backbone+1080p+WEB-DL+DD5+1+H+264-NTb.torrent

May 01, 2016 07:51:06 PM    Debug   

UnixLibCurlWebClient:GetBytes Returning OK => 307 bytes

May 01, 2016 07:51:06 PM    Debug   

UnixLibCurlWebClient:GetBytes(Url:http://hdts.ru/download.php?id=69aca296b4b894c1d99bad9f3c3f237bace36d5c&f=Criminal+Minds+S11E21+Devil's+Backbone+1080p+WEB-DL+DD5+1+H+264-NTb.torrent)

May 01, 2016 07:51:06 PM    Info    

ReflectedHttpActionDescriptor ExecuteAsync Invoking action 'Download(indexerID=hdtorrents, path=aHR0cDovL2hkdHMucnUvZG93bmxvYWQucGhwP2lkPTY5YWNhMjk2YjRiODk0YzFkOTliYWQ5ZjNjM2YyMzdiYWNlMzZkNWMmZj1DcmltaW5hbCtNaW5kcytTMTFFMjErRGV2aWwncytCYWNrYm9uZSsxMDgwcCtXRUItREwrREQ1KzErSCsyNjQtTlRiLnRvcnJlbnQ1, apikey=xxxxxx, file=Criminal Minds S11E21 Devil's Backbone 1080p WEB-DL DD5 1 H 264-NTb.torrent)'

May 01, 2016 07:51:06 PM    Info    

ApiControllerActionInvoker InvokeActionAsync Action='Download(indexerID=hdtorrents, path=aHR0cDovL2hkdHMucnUvZG93bmxvYWQucGhwP2lkPTY5YWNhMjk2YjRiODk0YzFkOTliYWQ5ZjNjM2YyMzdiYWNlMzZkNWMmZj1DcmltaW5hbCtNaW5kcytTMTFFMjErRGV2aWwncytCYWNrYm9uZSsxMDgwcCtXRUItREwrREQ1KzErSCsyNjQtTlRiLnRvcnJlbnQ1, apikey=xxxxx, file=Criminal Minds S11E21 Devil's Backbone 1080p WEB-DL DD5 1 H 264-NTb.torrent)'

May 01, 2016 07:51:06 PM    Info    

JackettAPINoCacheAttribute OnActionExecutingAsync 

May 01, 2016 07:51:06 PM    Info    

JackettAPINoCacheAttribute OnActionExecutingAsync Action filter for 'Download(String indexerID, String path, String apikey, String file)'

May 01, 2016 07:51:06 PM    Info    

HttpActionBinding ExecuteBindingAsync Model state is valid. Values: indexerID=hdtorrents, path=aHR0cDovL2hkdHMucnUvZG93bmxvYWQucGhwP2lkPTY5YWNhMjk2YjRiODk0YzFkOTliYWQ5ZjNjM2YyMzdiYWNlMzZkNWMmZj1DcmltaW5hbCtNaW5kcytTMTFFMjErRGV2aWwncytCYWNrYm9uZSsxMDgwcCtXRUItREwrREQ1KzErSCsyNjQtTlRiLnRvcnJlbnQ1, apikey=xxxxxx, file=Criminal Minds S11E21 Devil's Backbone 1080p WEB-DL DD5 1 H 264-NTb.torrent

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Parameter 'file' bound to the value 'Criminal Minds S11E21 Devil's Backbone 1080p WEB-DL DD5 1 H 264-NTb.torrent'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Binding parameter 'file'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Parameter 'apikey' bound to the value 'xxxxxxx'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Binding parameter 'apikey'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Parameter 'path' bound to the value 'aHR0cDovL2hkdHMucnUvZG93bmxvYWQucGhwP2lkPTY5YWNhMjk2YjRiODk0YzFkOTliYWQ5ZjNjM2YyMzdiYWNlMzZkNWMmZj1DcmltaW5hbCtNaW5kcytTMTFFMjErRGV2aWwncytCYWNrYm9uZSsxMDgwcCtXRUItREwrREQ1KzErSCsyNjQtTlRiLnRvcnJlbnQ1'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Binding parameter 'path'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Parameter 'indexerID' bound to the value 'hdtorrents'

May 01, 2016 07:51:06 PM    Info    

ModelBinderParameterBinding ExecuteBindingAsync Binding parameter 'indexerID'

May 01, 2016 07:51:06 PM    Info    

HttpActionBinding ExecuteBindingAsync 

May 01, 2016 07:51:06 PM    Info    

ApiControllerActionSelector SelectAction Selected action 'Download(String indexerID, String path, String apikey, String file)'

May 01, 2016 07:51:06 PM    Info    

ApiControllerActionSelector SelectAction 

May 01, 2016 07:51:06 PM    Info    

DownloadController ExecuteAsync 

May 01, 2016 07:51:06 PM    Info    

HttpControllerDescriptor CreateController Jackett.Controllers.DownloadController

May 01, 2016 07:51:06 PM    Info    

DefaultHttpControllerActivator Create Jackett.Controllers.DownloadController

May 01, 2016 07:51:06 PM    Info    

DefaultHttpControllerActivator Create 

May 01, 2016 07:51:06 PM    Info    

HttpControllerDescriptor CreateController 
```

I tested Sonarr with another episode that didn't have the apostrophe in the title and it worked fine. I can copy the URL listed above in the logs and open it in Firefox and it grabs the file fine. Jackett is running on a Debian box for me.

**Comment by JigSawFr** [comment]
I think this is related to provider.

**Comment by SaskiFX** [comment]
Any way I can check? Downloading another torrent without one worked from the same provider, and I can get the file in my browser.

**Comment by JigSawFr** [comment]
If you can download it from Jackett by entering URL, the problem may be related to Sonarr.

**Comment by SaskiFX** [comment]
I don't understand what you mean by download it from Jackett. Jackett fails to get the file from the torrent site, like in the logs up above.

```
May 01, 2016 07:51:06 PM    Error   

Error downloading hdtorrents http://hdts.ru/download.php?id=69aca296b4b894c1d99bad9f3c3f237bace36d5c&f=Criminal+Minds+S11E21+Devil's+Backbone+1080p+WEB-DL+DD5+1+H+264-NTb.torrent
```

But if I punch that link into firefox, it downloads fine.

**Comment by kaso17** [comment]
Fixed with 7835da2
